backend/app/services/adapters/task_kinds.py

```
@@ -95,23 +95,29 @@ def create_task_or_append(
                 expire_hours = settings.APPEND_CODE_TASK_EXPIRE_HOURS
-            if (
-                datetime.now() - existing_task.updated_at
-            ).total_seconds() > expire_hours * 3600:
-                raise HTTPException(
-                    status_code=400,
-                    detail=f"{task_type} task has expired. You can only append tasks within {expire_hours} hours after last update.",
-                )
+            
+            task_shell_source = (
+                task_crd.chat_shell.labels
+                and task_crd.chat_shell.labels.get("source") 
```

Ruff (0.14.7)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)